sdk/python/feast/permissions/security_manager.py

```diff
@@ -84,6 +85,45 @@ def assert_permissions(
         )
 
 
+def assert_permissions_to_update(
+    resource: FeastObject,
+    actions: Union[AuthzedAction, List[AuthzedAction]],
+    getter: Callable[[str, str, bool], FeastObject],
+    project: str,
+    allow_cache: bool = True,
+) -> FeastObject:
+    """
+    Verify if the current user is authorized to execute the requested actions on the given resource and also checks the authorization on
+    the existing resource, if any.
+
+    If no permissions are defined, the result is to deny the execution.
+
+    Args:
+        resource: The resources for which we need to enforce authorized permission.
+        actions: The requested actions to be authorized.
+        getter: The getter function used to retrieve the existing resource instance by name.
+        The signature must be `get_permission(self, name: str, project: str, allow_cache: bool)`
+        project: The project nane used in the getter function.
+        allow_cache: Whether to use cached data. Defaults to `True`.
+    Returns:
+        FeastObject: The original `resource`, if permitted.
+
+    Raises:
+        PermissionError: If the current user is not authorized to execute all the requested actions on the given resource or on the existing one.
+    """
+    resource_to_update = assert_permissions(resource=resource, actions=actions)
```

Comment:
Makes sense (and it has changed to `DESCRIBE` instead of `READ`, after a long discussion 😉 ).
And to take it a step further, I’d also hide the `actions` parameter entirely, since the actions are already implied by the function.

Reply:
and here we go.. I'd have totally mixed them up 😄 +1 for dropping actions param as well